python/link_prediction.py

```diff
@@ -0,0 +1,771 @@
+
+    Args:
+        ctx (mgp.ProcCtx, optional): Reference to the process execution.
+
+    Returns:
+        mgp.Record: It returns performance metrics obtained during the training on the training and validation dataset.
+    """
+    # Get global context
+    global training_results, validation_results, predictor, model, graph, new_to_old, old_to_new
+
+    # Reset parameters of the old training
+    _reset_train_predict_parameters()
+
+    # Get some data
+    graph, new_to_old, old_to_new = _get_dgl_graph_data(ctx)  # dgl representation of the graph and dict new to old index
+
+    # Check if there are no edges in the dataset, assume that it cannot learn effectively without edges. E2E handling.
+    if graph.number_of_edges() == 0:
+        raise Exception("No edges in the dataset. ")
+
+    # TEST: Currently disabled
+    _conversion_to_dgl_test(graph=graph, new_to_old=new_to_old, ctx=ctx, node_features_property=link_prediction_parameters.node_features_property)
+
+    """ Train g is a graph which has removed test edges. Others are positive and negative train and test graphs
+    """
+
+    # Insert in the hidden_features_size structure if needed
+    _load_feature_size(graph.ndata[link_prediction_parameters.node_features_property].shape[1])
+
+    # Split the data
+    train_g, train_pos_g, train_neg_g, val_pos_g, val_neg_g = preprocess(graph, link_prediction_parameters.split_ratio)
+
+    # Create a model
+    model = create_model(
+        layer_type=link_prediction_parameters.layer_type,
+        hidden_features_size=link_prediction_parameters.hidden_features_size,
+        aggregator=link_prediction_parameters.aggregator,
+        attn_num_heads=link_prediction_parameters.attn_num_heads,
+    )
+    # Create a predictor
+    predictor_hidden_size = link_prediction_parameters.hidden_features_size[-1]
+    
+
+    predictor = create_predictor(
+        predictor_type=link_prediction_parameters.predictor_type,
+        predictor_hidden_size=predictor_hidden_size,
+    )
+
+    # Create an optimizer
+    optimizer = create_optimizer(
+        optimizer_type=link_prediction_parameters.optimizer,
+        learning_rate=link_prediction_parameters.learning_rate,
+        model=model,
+        predictor=predictor,
+    )
+
+    # Collect training and validation results from utils model
+    training_results, validation_results = inner_train(
+        model,
+        predictor,
+        optimizer,
+        link_prediction_parameters.num_epochs,
+        link_prediction_parameters.node_features_property,
+        link_prediction_parameters.console_log_freq,
+        link_prediction_parameters.checkpoint_freq,
+        link_prediction_parameters.metrics,
+        link_prediction_parameters.tr_acc_patience,
+        link_prediction_parameters.context_save_dir,
+        train_g,
+        train_pos_g,
+        train_neg_g,
+        val_pos_g,
+        val_neg_g,
+    )
```

Comment:
Common Python practice is to have usually optional arguments in the end. This means that negative and positive graphs should be up front, while parameters such as load directory and log frequency come at the end.